handlers/java_handler/src/main/java/mercator/MavenUtils.java

```diff
@@ -79,8 +82,47 @@ public static Document getParsedExpandedPom(File pomFile) {
             return null;
         }
 
-        Document parsedPom = readFileAsDocument(resolvedPom);
-        return parsedPom;
+        try {
+            File resolvedPomFixed = removeDuplicateXMLTag(resolvedPom);
+            Document parsedPom = readFileAsDocument(resolvedPomFixed);
+            return parsedPom;
+        } catch (java.io.IOException ex) {
+            return null;
+        }
+
+    }
+
+    /**
+     * Remove duplicate XML start tag from given XML file.
+     * @param resolvedPom
+     * @return File handle having only one XML start tag
+     * @throws IOException
+     */
+    public static File removeDuplicateXMLTag(File resolvedPom) throws IOException {
+        File resolvedPomFixed;
+        resolvedPomFixed = File.createTempFile("resolvedpom-fixed", ".xml");
+        resolvedPomFixed.deleteOnExit();
+
+        Scanner sc = new Scanner(new FileInputStream(resolvedPom));
+        String xmlTag = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>";
```

Comment:
Does this cover all variants of a beginning of XML? What if there will be different encoding? Maybe some redex patter would be more suitable here.

Anyway - don't you need to remove also comments, which are there? I think that it's not really valid XML if it does not start with `<?xml version...`. Correct me If I am wrong.

Reply:
+1, the valid XML should start with prolog: <?xml version=\"1.0\" encoding=\"UTF-8\"?>"